tests/Makefile

```diff
@@ -57,6 +57,10 @@ else
 DLEXT ?= so
 EXEXT=
 SHARED_SUFFIX=
+
+ifdef PTHREAD
```

Comment:
Note that `ifdef PTHREAD` is true when invoked with `make PTHREAD=0`

I would marginally prefer a variable like `THREADING` rather than `PTHREAD`, so we can do

```
make THREADING=pthread
```

```
ifeq ($(THREADING),pthread)
:
```

Reply:
Ideally we wouldn't change how `ssl_pthread_server` can be built. But it's better to break some builds of a sample program than all builds on some platforms.